pkg/xds/envoy/clusters/v3/outlier_detection_configurer.go

```diff
@@ -0,0 +1,87 @@
+package clusters
+
+import (
+	envoy_cluster "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	"github.com/golang/protobuf/ptypes/wrappers"
+
+	mesh_core "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+)
+
+type OutlierDetectionConfigurer struct {
+	CircuitBreaker *mesh_core.CircuitBreakerResource
+}
+
+var _ ClusterConfigurer = &OutlierDetectionConfigurer{}
+
+func (c *OutlierDetectionConfigurer) Configure(cluster *envoy_cluster.Cluster) error {
+	if c.CircuitBreaker == nil {
+		return nil
+	}
+
+	cluster.OutlierDetection = &envoy_cluster.OutlierDetection{
+		Interval:                       c.CircuitBreaker.Spec.GetConf().Interval,
+		BaseEjectionTime:               c.CircuitBreaker.Spec.GetConf().BaseEjectionTime,
+		MaxEjectionPercent:             c.CircuitBreaker.Spec.GetConf().MaxEjectionPercent,
+		SplitExternalLocalOriginErrors: c.CircuitBreaker.Spec.GetConf().SplitExternalAndLocalErrors,
+	}
+	c.configureTotalErrorDetector(cluster.OutlierDetection)
+	c.configureGatewayErrorDetector(cluster.OutlierDetection)
+	c.configureLocalErrorDetector(cluster.OutlierDetection)
+	c.configureStandardDeviationDetector(cluster.OutlierDetection)
+	c.configureFailureDetector(cluster.OutlierDetection)
+	return nil
+}
+
+func (c *OutlierDetectionConfigurer) configureTotalErrorDetector(outlierDetection *envoy_cluster.OutlierDetection) {
+	if total := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetTotalErrors(); total != nil {
+		outlierDetection.Consecutive_5Xx = total.GetConsecutive()
+		outlierDetection.EnforcingConsecutive_5Xx = &wrappers.UInt32Value{Value: 100}
+	} else {
+		outlierDetection.EnforcingConsecutive_5Xx = &wrappers.UInt32Value{Value: 0}
+	}
+}
+
+func (c *OutlierDetectionConfigurer) configureGatewayErrorDetector(outlierDetection *envoy_cluster.OutlierDetection) {
+	if gateway := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetGatewayErrors(); gateway != nil {
+		outlierDetection.ConsecutiveGatewayFailure = gateway.GetConsecutive()
+		outlierDetection.EnforcingConsecutiveGatewayFailure = &wrappers.UInt32Value{Value: 100}
+	} else {
+		outlierDetection.EnforcingConsecutiveGatewayFailure = &wrappers.UInt32Value{Value: 0}
+	}
+}
+
+func (c *OutlierDetectionConfigurer) configureLocalErrorDetector(outlierDetection *envoy_cluster.OutlierDetection) {
+	if local := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetLocalErrors(); local != nil {
```

Comment:
According to https://github.com/kumahq/kuma/blob/ff2e4f2eab0d5da3f08397692a4c69c2b9a9513d/api/mesh/v1alpha1/circuit_breaker.pb.go#L175-L180
`c.CircuitBreaker.Spec.GetConf().GetDetectors()` can return nil